converse.js

```diff
@@ -634,11 +635,35 @@
             },this), 200));
         };
 
+        this.ping = function (jid, timeout, success, error){
+            if (typeof jid === 'undefined'     || jid == null)     { jid = this.bare_jid; }
+                if (typeof timeout === 'undefined' || timeout == null) { timeout = 45; }
+                if (typeof success === 'undefined' ) { success = null; }
+                if (typeof error === 'undefined' ) { error = null; }
+                if (this.connection) this.connection.ping.ping( jid, success, error, timeout );
+        };
```

Comment:
We should add add a callback method for `error` so that we can call `converse.reconnect` or `converse.renderLoginPanel`.

The code would be the same as here: https://github.com/jcbrand/converse.js/blob/a69ceba813c780943977df771329cb6687c99e3e/converse.js#L483

So I think we can refactor that out into a separate method and make it available as the error handler for `converse.ping`.